### ISSUE TITLE
Consul Server restart results in failure

### ISSUE THREAD

**Issue by saswatp** [CiscoCloud/mantl#566]
Steps to reproduce : 

1. Stop consul service in control node 1
2. Stop Consul service in control node 2
3. Start Consul service in control node1 
4. Start Consul service in control node2 

After this point consul doesn't come up and the entire cluster is unusable. 
service consul status throws error (no cluster leader) : 
```
service consul status
Redirecting to /bin/systemctl status  consul.service
consul.service - Consul is a tool for service discovery and configuration. Consul is distributed, highly available, and extremely scalable.
   Loaded: loaded (/usr/lib/systemd/system/consul.service; enabled)
   Active: active (running) since Mon 2015-07-13 20:55:17 UTC; 4min 17s ago
     Docs: http://www.consul.io
 Main PID: 22302 (consul)
   CGroup: /system.slice/consul.service
           └─22302 /usr/bin/consul agent -config-dir=/etc/consul -data-dir=/var/lib/consul

Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: 2015/07/13 20:59:34 [ERR] http: Request /v1/health/service/mesos-leader?index=931&wait=60000ms, error: No cluster leader
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: http: Request /v1/health/service/mesos-leader?index=931&wait=60000ms, error: No cluster leader
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: 2015/07/13 20:59:34 [ERR] http: Request /v1/health/service/chronos?index=931&wait=60000ms, error: No cluster leader
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: http: Request /v1/health/service/chronos?index=931&wait=60000ms, error: No cluster leader
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: 2015/07/13 20:59:34 [ERR] http: Request /v1/health/service/mesos-follower?index=931&wait=60000ms, error: No cluster leader
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: http: Request /v1/health/service/mesos-follower?index=931&wait=60000ms, error: No cluster leader
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: 2015/07/13 20:59:34 [ERR] http: Request /v1/health/service/marathon?index=931&wait=60000ms, error: No cluster leader
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: http: Request /v1/health/service/marathon?index=931&wait=60000ms, error: No cluster leader
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: 2015/07/13 20:59:34 [WARN] agent: Check 'service:chronos' is now critical
Jul 13 20:59:34 mi-control-02.novalocal consul[22302]: agent: Check 'service:chronos' is now critical

```

**Comment by stevendborrelli** [comment]
Are you only running two nodes? A restart of a node in a two server cluster would cause a quorum issue.

**Comment by saswatp** [comment]
Same problem seen with three nodes control cluster as well.

**Comment by stevendborrelli** [comment]
Are you taking more than one server down at a time?

**Comment by saswatp** [comment]
Not simultaneously . Take down one by one. However, if I restart control VMs it could be simultaneously. This is a valid use case from CCS point of view. What I heard is that multiple VMs could reboot randomly and simultaneously , sometimes.

**Comment by vimvim** [comment]
The same issue. 3 control nodes rebooted almost simultaneously because of the DC failure. Consul  unable to start after reboot, mesos master unable to start because unable to resolve zookeeper.service.consul, zookeeper unable to start because unable to contact consul:
2015/08/05 13:36:16 [WARN] (runner) disabling consul SSL verification
2015/08/05 13:36:36 [ERR] (view) "service(zookeeper [passing,warning])" health services: error fetching: Get https://consul.service.consul:8500/v1/health/service/zookeeper?wait=60000ms: dial tcp: lookup consul.service.consul on 172.16.0.50:53: read udp 172.16.0.50:53: i/o timeout
2015/08/05 13:36:36 [ERR] (runner) watcher reported error: health services: error fetching: Get https://consul.service.consul:8500/v1/health/service/zookeeper?wait=60000ms: dial tcp: lookup consul.service.consul on 172.16.0.50:53: read udp 172.16.0.50:53: i/o timeout
2015/08/05 13:37:01 [ERR] (view) "service(zookeeper [passing,warning])" health services: error fetching: Get https://consul.service.consul:8500/v1/health/service/zookeeper?wait=60000ms: dial tcp: lookup consul.service.consul on 172.16.0.50:53: read udp 172.16.0.50:53: i/o timeout
2015/08/05 13:37:01 [ERR] (runner) watcher reported error: health services: error fetching: Get https://consul.service.consul:8500/v1/health/service/zookeeper?wait=60000ms: dial tcp: lookup consul.service.consul on 172.16.0.50:53: read udp 172.16.0.50:53: i/o timeout

Cluster is completely down and can't be started.

**Comment by BrianHicks** [comment]
Looks like this will be solved on our end by #763. It probably should be filed as a bug upstream though. Does anyone know if there's an open ticket with Hashicorp?

**Comment by siddharthist** [comment]
Closed in #948
